contracts/voting-verifier/src/query.rs

```diff
@@ -76,18 +90,18 @@ fn verification_status<T: PartialEq + std::fmt::Debug>(
                 Some(Vote::SucceededOnChain) => VerificationStatus::SucceededOnSourceChain,
                 Some(Vote::FailedOnChain) => VerificationStatus::FailedOnSourceChain,
                 Some(Vote::NotFound) => VerificationStatus::NotFoundOnSourceChain,
-                None if is_finished(&poll) => VerificationStatus::FailedToVerify,
+                None if is_finished(&poll, cur_block_height) => VerificationStatus::FailedToVerify,
                 None => VerificationStatus::InProgress,
             }
         }
         None => VerificationStatus::Unknown,
     }
 }
 
-fn is_finished(poll: &state::Poll) -> bool {
+fn is_finished(poll: &state::Poll, cur_block_height: u64) -> bool {
     match poll {
         state::Poll::Messages(poll) | state::Poll::ConfirmVerifierSet(poll) => {
-            poll.status == PollStatus::Finished
+            cur_block_height >= poll.expires_at
```

Comment:
```suggestion
            poll.status == PollStatus::Finished | cur_block_height >= poll.expires_at
```
If a poll completes before expiry it should also count as finished

Reply:
it's actually not possible for a poll to be in the finished state prior to expiration at the moment

Reply:
that's fine, I would still like to make it explicit instead of relying on implicit state checks. That could potentially change in the future and then this code breaks without us knowing. I'm also fine if you want to extract it out into a function that is used across the code base. I just don't want multiple checks across the codebase that imply state from circumstantial evidence. They are too easy to miss when changing behaviour and we have an explicit state to use. Frankly, I would prefer it even more if we for example change `poll.status` into a function that takes the current block and returns the correct state on it's own. Then we could add a `PollStatus::Expired` and maybe rename `PollStatus::Finished` to make the difference clear, so all the status logic is directly tied to the poll instead of distributed across the code

Reply:
I changed the status into a function, lmk what you think. It was a bit weird because we still need to actually modify the struct somehow when `finish` is called, but whether a poll is in progress or expired is just determined dynamically.